website/pages/blog/building-cloudquery.md

```diff
@@ -0,0 +1,104 @@
+---
+title: "Building CloudQuery: High Performance Data Integration Framework in Go"
+tag: engineering
+date: 2022/11/28
+description: >-
+  Deep dive on how we built CloudQuery and the design decisions we made along the way.
+author: yevgenypats
+---
+
+
+import { BlogHeader } from "../../components/BlogHeader"
+
+<BlogHeader/>
+
+Data Integration is a 30+ year-old problem of extracting data from APIs, normalizing it and loading it into various different destinations such as databases, data lakes and files. In the last 20 years the problem has become acute, especially in the marketing and business analytics space, but recently also in the infrastructure world, with the explosion of cloud vendors and infrastructure services.
+
+Even though data integration is an old problem and will never be [completely](https://twitter.com/mattrickard/status/1542193426979909634) solved (at least while SaaS services don’t get better at having consistent machine readable APIs), we believe there is a big opportunity to address this problem at scale and achieve a new local optimum, and it can be done right now.
+
+Addressing this problem comes with significant challenges. In this post, I’ll go over these challenges, why they’re hard, and the solutions we landed on in CloudQuery’s design.
+
+# Engineering Requirements and Challenges
+
+Before we go over the technical and architecture decisions we made in CloudQuery, let’s cover the main challenges and requirements of an ELT framework:
+
+- **API Coverage:** Scaling and supporting the number of APIs covered in an efficient manner (i.e less developers, more APIs supported)--leveraging the community is key.
+- **Scaling Destinations:** Unlike sources, destinations are finite but the right architecture should ensure source and destinations are decoupled correctly so new destinations will work out-of-the-box and won't require any changes in sources (otherwise development work will grow by `n*m` where `n` is number of source plugins and `m` is number of destination plugins).
```

Comment:
```suggestion
- **Scaling Destinations:** Unlike sources, destinations are finite but the right architecture should ensure sources and destinations are decoupled correctly so new destinations will work out-of-the-box and won't require any changes in sources (otherwise development work will grow by `n*m` where `n` is number of source plugins and `m` is number of destination plugins).
```